ConfigSpace/util.pyx

```diff
@@ -436,3 +436,185 @@ def fix_types(configuration: dict,
             else:
                 raise TypeError("Unknown hyperparameter type %s" % type(param))
     return configuration
+
+cimport cython
+@cython.boundscheck(True)  # Activate bounds checking
+@cython.wraparound(True)  # Activate negative indexing
+def generate_grid(configuration_space: ConfigurationSpace,
+                 num_steps_dict: Union[None,  Dict[str, int]] = None,
+                 ) -> List[Configuration]:
+    """
+    Generates a grid of Configurations for a given ConfigurationSpace. Can be used, for example, for grid search.
+
+    Parameters
+    ----------
+    configuration_space: :class:`~ConfigSpace.configuration_space.ConfigurationSpace`
+        The Configuration space over which to create a grid of HyperParameter Configuration values. It knows the types for all parameter values.
+
+    num_steps_dict: dict
+        A dict containing the number of points to divide the grid side formed by Hyperparameters which are either of type UniformFloatHyperparameter or type UniformIntegerHyperparameter. The keys in the dict should be the names of the corresponding Hyperparameters and the values should be the number of points to divide the grid side formed by the corresponding Hyperparameter in to.
+
+    Returns
+    -------
+    list
+        List containing Configurations. It is a cartesian product of tuples of HyperParameter values. Each tuple lists the possible values taken by the corresponding HyperParameter. Within the cartesian product, in each element, the ordering of HyperParameters is the same for the OrderedDict within the ConfigurationSpace.
+    """
+
+    def get_value_set(num_steps_dict, hp_name):
+        '''
+        Gets values along the grid for a particular hyperparameter.
+
+        Uses the num_steps_dict to determine number of grid values for UniformFloatHyperparameter and UniformIntegerHyperparameter. If these values are not present in num_steps_dict, the quantization, q, objects of these classes will be used to divide the grid. NOTE: If q is None, this results in np.arange using None for its step size parameter which results in a quantization of 1.

```

Comment:
I'm wondering whether it would make more sense to fail if neither the number of steps nor q is given instead of assuming the user meant to set the value to 1. What do you think?

Reply:
Yes, I had thought about throwing a warning, especially if the number of possible grid points is "too many". But probably, an Error is better. Added that now.